editor/plugins/tile_map_editor_plugin.cpp

```diff
@@ -532,12 +532,6 @@ PoolVector<Vector2> TileMapEditor::_bucket_fill(const Point2i &p_start, bool era
 		return PoolVector<Vector2>();
 	}
 
-	for (int i = ids.size() - 1; i >= 0; i--) {
-		if (ids[i] == prev_id) {
```

Comment:
I think the check should be kept in the `ids.size() == 1` scenario (like most bucket fill tools in painting applications, if you try to paint a color on the same color, nothing should happen).